src/dmd/root/errno.d

```diff
@@ -0,0 +1,73 @@
+/**
+ * Pure wrapper of `errno`.
+ *
+ * Compiler implementation of the D programming language
+ * http://dlang.org
+ *
+ * Copyright: Copyright (C) 1999-2019 by The D Language Foundation, All Rights Reserved
+ * Authors:   Walter Bright, http://www.digitalmars.com
+ * License:   $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:    $(LINK2 https://github.com/dlang/dmd/blob/master/src/dmd/root/errno.d, root/_errno.d)
+ * Documentation:  https://dlang.org/phobos/dmd_root_errno.html
+ * Coverage:    https://codecov.io/gh/dlang/dmd/src/master/src/dmd/root/errno.d
+ */
+module dmd.root.errno;
+
+static import core.stdc.errno;
+
+/**
+ * Restores `errno` after calling `func`.
+ *
+ * Params:
+ *  func = the action to perform before `errno` is restored
+ *
+ * Returns: whatever `func` returns
+ */
+package auto restoreErrno(alias func)()
```

Comment:
I know its not necessary but since it will be inferred, but the whole point of these PRs is to make stuff pure, it would be good to make that blindingly obvious by adding `pure` on this function.

Otherwise looks good.

Reply:
If I make the function `pure` then that forced the lambda to be `pure` as well. I think that's an unnecessary restriction.

I can make a `pure` unit test if that helps?